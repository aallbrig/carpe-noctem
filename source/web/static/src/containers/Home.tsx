import * as React from 'react';
import { connect, MapStateToProps, MapDispatchToPropsFunction } from 'react-redux';
import { Row, Col, Panel } from 'react-bootstrap';

interface IHomeContainerProps extends React.Props<any> {};
export class HomeContainer extends React.Component<IHomeContainerProps, void> {
  public render() {
    return (
      <Row>
        <Col xs={12}>
          <h4 className='text-center'> TODO: Place Hero Promotional Image Here For Game </h4>
        </Col>
        <Col xs={12}>
          <Panel header={'Panel Header'} footer={'Panel Footer'}>
            Maybe this panel can be used for something.
          </Panel>
        </Col>
      </Row>
    );
  };
};

const mapStateToProps: MapStateToProps<IHomeContainerProps, {}> = () => ({});
const mapDispatchToProps: MapDispatchToPropsFunction<{}, {}> = () => ({});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContainer);
