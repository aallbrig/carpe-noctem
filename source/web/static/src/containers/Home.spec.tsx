import * as React from 'react';
import { HomeContainer } from './Home';
import { connect } from 'react-redux'
import * as renderer from 'react-test-renderer';

test('Home Container', () => {
  const component = renderer.create(
    <HomeContainer />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});