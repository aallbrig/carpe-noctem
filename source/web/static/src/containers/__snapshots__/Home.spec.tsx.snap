// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12"
  >
    <h4
      className="text-center"
    >
       TODO: Place Hero Promotional Image Here For Game 
    </h4>
  </div>
  <div
    className="col-xs-12"
  >
    <div
      className="panel panel-default"
      id={undefined}
    >
      <div
        className="panel-heading"
      >
        Panel Header
      </div>
      <div
        className="panel-body"
      >
        Maybe this panel can be used for something.
      </div>
      <div
        className="panel-footer"
      >
        Panel Footer
      </div>
    </div>
  </div>
</div>
`;
